### PR TITLE
Reworks sampling for MACRO.

### PR DESCRIPTION
**Pull request by umbernhard**
**Description**

Swapping out consistent sampling for something that doesn't eat all of our memory

**Testing**

snapshots were updated. 

**Progress**

Need to resolve typing issues around ticket numbers no longer being here.

### CODE REVIEW

File: server/audit_math/sampler.py [hunk 1]
```diff
@@ -110,55 +110,22 @@ def draw_ppeb_sample(
...
-
-    return sample
+    weighted_errors = [
+        macro.compute_max_error(batch_results[batch], contest) / U
+        for batch in batch_results
+    ]
+
+    return generator.choice(
+        list(batch_results.keys()), sample_size + num_sampled, p=weighted_errors
+    )
```
Comment (jonahkagan) [inline server/audit_math/sampler.py:131]: do we need to cut out the first `num_sampled` from this result?

File: server/audit_math/sampler.py [hunk 2]
```diff
@@ -110,55 +110,22 @@ def draw_ppeb_sample(
...
-    for item in faux_sample:
-        sample.append((item[0], item[1][0], item[2]))
-
-    return sample
+    weighted_errors = [
+        macro.compute_max_error(batch_results[batch], contest) / U
+        for batch in batch_results
+    ]
+
+    return generator.choice(
```
Comment (jonahkagan) [inline server/audit_math/sampler.py:129]: maybe specify `replace=True` just to be sure - it's not totally clear that it's the default

Comment (umbernhard) [reply server/audit_math/sampler.py:129]: Good point!

File: server/audit_math/sampler.py (hunk 1 above)
Comment (umbernhard) [reply server/audit_math/sampler.py:131]: Yep, you're right.

**Review by jonahkagan — APPROVED** [review]

**Pull request merged**